pkg/apps/registry/instantiate/rest.go

```diff
@@ -300,20 +306,27 @@ func decodeFromLatestDeployment(config *appsapi.DeploymentConfig, rn kcoreclient
 	if config.Status.LatestVersion == 0 {
 		return config, nil
 	}
-
-	latestDeploymentName := appsinternalutil.LatestDeploymentNameForConfig(config)
+	externalConfig := &appsv1.DeploymentConfig{}
+	if err := legacyscheme.Scheme.Convert(config, externalConfig, nil); err != nil {
+		return nil, err
+	}
+	latestDeploymentName := appsutil.LatestDeploymentNameForConfig(externalConfig)
 	deployment, err := rn.ReplicationControllers(config.Namespace).Get(latestDeploymentName, metav1.GetOptions{})
 	if err != nil {
 		// If there's no deployment for the latest config, we have no basis of
 		// comparison. It's the responsibility of the deployment config controller
 		// to make the deployment for the config, so return early.
 		return nil, err
 	}
-	decoded, err := appsinternalutil.DecodeDeploymentConfig(deployment)
+	decoded, err := appsutil.DecodeDeploymentConfig(deployment)
```

Comment:
@deads2k this is questionable... I really don't want to carry a util to decode the deployment config to internal version... Is it acceptable to have just external helper and then do the conversion afterwards?

Reply:
(I don't think the instantiate API endpoint is something that heavily used and will cause performance hit.